pandas/core/algorithms.py

```diff
@@ -708,6 +708,10 @@ def value_counts(values, sort=True, ascending=False, normalize=False,
 
     if sort:
         result = result.sort_values(ascending=ascending)
+    elif bins is None:
+        uniq = unique(values)
+        if not isinstance(result.index, CategoricalIndex):
```

Comment:
why is this check needed? (or maybe it just needs to be for an *ordered* categorical)

Reply:
The issue was the test `TestCategoricalSeriesAnalytics.test_value_counts`:
```
cats = Categorical(list('abcccb'), categories=list('cabd'))
s = Series(cats, name='xxx')
res = s.value_counts(sort=False)
```
which returns `0` for the `d`category as well, which is not in the `unique(values)`. Is there another possibility to get access to that initial categorical from the index?